connectors/kafka/src/main/java/forklift/producers/KafkaForkliftProducer.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    @Override
+    public String send(Object message) throws ProducerException {
+        if (message instanceof SpecificRecord) {
+            return sendAvroMessage((SpecificRecord)message);
+        } else {
+            String json;
+            try {
+                json = mapper.writeValueAsString(message);
+            } catch (JsonProcessingException e) {
+                throw new ProducerException("Error creating Kafka Message", e);
+            }
+            return sendForkliftWrappedMessage(json, null);
+        }
+    }
+
+    @Override
+    public String send(Map<String, String> message) throws ProducerException {
+        return sendForkliftWrappedMessage(this.formatMap(message), null);
+    }
+
+    @Override
+    public String send(Map<Header, Object> headers, Map<String, String> properties, ForkliftMessage message)
+                    throws ProducerException {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public String send(Map<String, String> properties, ForkliftMessage message)
+                    throws ProducerException {
+        return this.sendForkliftWrappedMessage(message.getMsg(), properties);
+    }
+
+    @Override
+    public Map<String, String> getProperties() throws ProducerException {
+        return this.properties;
+    }
+
+    @Override
+    public void setProperties(Map<String, String> properties) throws ProducerException {
+        this.properties = properties;
+    }
+
+    @Override
+    public void setHeaders(Map<Header, Object> haeders) {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public Map<Header, Object> getHeaders() {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public void close() throws IOException {
+        //do nothing, the passed in KafkaProducer may be used elsewhere and should be closed by the KafkaController
+    }
+
+    private String sendForkliftWrappedMessage(String message, Map<String, String> messageProperties) throws ProducerException {
+        GenericRecord avroRecord = new GenericData.Record(forkliftSchema);
+        avroRecord.put(SCHEMA_FIELD_NAME_VALUE, message);
+        messageProperties = messageProperties == null ? new HashMap<>() : new HashMap<>(messageProperties);
+        //add the producer level properties but do not overwrite message level properties
+        for (Map.Entry<String, String> entry : this.properties.entrySet()) {
+            messageProperties.putIfAbsent(entry.getKey(), entry.getValue());
+        }
+        avroRecord.put(SCHEMA_FIELD_NAME_PROPERTIES, this.formatMap(messageProperties));
+        ProducerRecord record = new ProducerRecord<>(topic, null, avroRecord);
+        try {
+            RecordMetadata result = (RecordMetadata)kafkaProducer.send(record).get();
```

Comment:
Can we stay asynchronous? Batching messages together is one of the core things that makes Kafka fast, so sending them synchronously one-at-a-time is missing out on a lot.

If you don't want the client of this code to deal with `RecordMetadata`, there are better ways than just making this synchronous (another callback, returning a different future, etc).